src/ruby/puppet-server-lib/puppet/server/http_client.rb

```diff
@@ -35,8 +34,8 @@ def initialize(server, port, options = {})
 
     @server = server
     @port = port
-    @use_ssl = options[:use_ssl]
-    @protocol = @use_ssl ? "https" : "http"
+    self.class.use_ssl = options[:use_ssl]
```

Comment:
Also, I think we should add some sort of test for "http" and "https" calls off the same client to the unit tests below to prove that this will work.  :)